src/render/html.jl

```diff
@@ -153,8 +165,39 @@ function footerhtml(io::IO, doc::Metadata, config::Config)
     """)
 end
 
-function wrap(fn::Function, io::IO, tag::String, attributes::String = "")
+function wrap(fn::Function, io::IO, tag::AbstractString, attributes::AbstractString = "")
     println(io, "<", tag, " ", attributes, ">")
     fn()
     println(io, "</", tag, ">")
 end
+
+## Utils
+
+# Convert's a string to a valid html id
+function generate_html_id(s::AbstractString)
+    # http://www.w3.org/TR/html4/types.html#type-id
+    # ID tokens must begin with a letter ([A-Za-z]) and may be followed by any number of letters, 
+    # digits ([0-9]), hyphens ("-"), underscores ("_"), colons (":"), and periods (".").
+    valid_chars = Set(Char['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+                           'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+                           'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+                           'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+                           '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 
+                           '-', '_', ':', '.'])
+
+    replace_chars = Set(Char[' ', '!', '"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', 
+                             '/', ';', '<', '=', '>', '?', '@', '[', ']', '^', '`', '{', 
+                             '¦', '}', '~'])                 
+    id = ""
+    for c in s
+        if c in valid_chars 
+            id = id * lowercase(string(c)) 
+        elseif c in replace_chars
+            id = id * "_"
+        else
+            id = id * string(Int(c))
+        end
+    end
```

Comment:
Could use an `IOBuffer` here instead of `id = ""`, append strings to it using `write`, and then get the final result with `takebuf_string`. Should be faster for longer ids since it won't create a new string on every loop iteration.